templates/ai/content/ai-basics/_ai-basics-carousel.html

```diff
@@ -0,0 +1,27 @@
+{% set slides = [
+  { "alt": "Tokenization Visualization","title": "Tokenization", "description": "The LLM starts by breaking down the input text, or 'prompt', into smaller pieces known as tokens.", "image": "/img/ai/carousels/how-llm-works/tokenization.png"},
+  { "alt": "Embedding Visualization","title": "Embedding", "description": "Next, each token is converted into a numerical representation through a process called embedding", "image": "/img/ai/carousels/how-llm-works/embedding.png"},
+  { "alt": "Self-attention Visualization","title": "Self-attention", "description": "The model then adds additional information to these vectors to account for the order of words in the input. Self attention mechanisms allow the model to create context-aware representations of each word.", "image": "/img/ai/carousels/how-llm-works/self-attention.png"},
+  { "alt": "Decoding Visualization","title": "Decoding", "description": "The LLM Decoder will take the output and use it to predict the next token in the sequence.", "image": "/img/ai/carousels/how-llm-works/decoding.png"},
+  { "alt": "Output Visualization","title": "Output", "description": "Finally, the model generates a response.", "image": "/img/ai/carousels/how-llm-works/output.png"}
+]
+%}
```

Comment:
This looks the same as https://github.com/skyfallsin/MIECO/blob/4903eda1110f1fb46f453ba378b209a45d8d79da/pages/ai/content/ai-basics/_ai-basics-carousel.html#L1-L8

Not sure if there is an easy way to share the contents between the files to keep them DRYer.
In fact, I think the two files are exactly the same. Interesting.

Also, one day we should talk about the Wonderful World of Eleventy and if it'd make this pipeline a lot easier (since it has native support for Nunjucks+Liquid and a bunch of other nice things baked in).

Reply:
Also, since I was I was looking at this for something else, Nunjucks [`{% macro %}`](https://mozilla.github.io/nunjucks/templating.html#macro)s might be a nice option. We could define the carousel code in one place, then just [`{% import %}`](https://mozilla.github.io/nunjucks/templating.html#import) the carousel `macro` in the templates where we need carousels (passing in the `slides[]` array as a parameter).

Reply:
@pdehaan  -  the two files you are referencing are the same files - one is just the product of running the build script @skyfallsin has created I think (and seems to be wrapping lines vs keeping them continuous).

Agreed on some level of optimized, out of the box tooling being helpful here though.